ovos_plugin_manager/templates/tts.py

```diff
@@ -44,272 +45,25 @@
-            self.attached_tts.append(tts)
-
-    def detach_tts(self, tts):
-        """Remove TTS from cache check."""
-        if tts in self.attached_tts:
-            self.attached_tts.remove(tts)
-
-    def get_attached_tts(self, tts_id=None):
-        tts_id = tts_id or self.active_tts
-        if not tts_id:
-            return
-        for tts in self.attached_tts:
-            if hasattr(tts, "tts_id"):
-                # opm plugin
-                if tts.tts_id == tts_id:
-                    return tts
-
-        for tts in self.attached_tts:
-            if not hasattr(tts, "tts_id"):
-                # non-opm plugin
-                if tts.tts_name == tts_id:
-                    return tts
-
-    def clear_queue(self):
-        """Remove all pending playbacks."""
-        while not self.queue.empty():
-            self.queue.get()
-        try:
-            self.p.terminate()
-        except Exception:
-            pass
-
-    def begin_audio(self, message=None):
-        """Perform beginning of speech actions."""
-        if self.bus:
-            message = message or Message("speak")
-            self.bus.emit(message.forward("recognizer_loop:audio_output_start"))
-        else:
-            LOG.warning("Speech started before bus was attached.")
-
-    def end_audio(self, listen, message=None):
-        """Perform end of speech output actions.
-        Will inform the system that speech has ended and trigger the TTS's
-        cache checks. Listening will be triggered if requested.
-        Args:
-            listen (bool): True if listening event should be emitted
-        """
-        if self.bus:
-            # Send end of speech signals to the system
-            message = message or Message("speak")
-            self.bus.emit(message.forward("recognizer_loop:audio_output_end"))
-            if listen:
-                self.bus.emit(message.forward('mycroft.mic.listen'))
-        else:
-            LOG.warning("Speech started before bus was attached.")
-
-    def on_start(self, message=None):
-        self.blink(0.5)
-        if not self._processing_queue:
-            self._processing_queue = True
-            self.begin_audio(message)
-
-    def on_end(self, listen=False, message=None):
-        if self._processing_queue:
-            self.end_audio(listen, message)
-            self._processing_queue = False
-        # Clear cache for all attached tts objects
-        # This is basically the only safe time
-        for tts in self.attached_tts:
-            tts.cache.curate()
-        self.blink(0.2)
-
-    def _play(self):
+    """ PlaybackThread moved to ovos_audio.playback
+    standalone plugin usage should rely on self.get_tts
+    ovos-audio relies on self.execute and needs this class
+
+    this class was only in ovos-plugin-manager in order to
+    patch usage of our plugins in mycroft-core"""
+    def __new__(self, *args, **kwargs):
+        LOG.warning("PlaybackThread moved to ovos_audio.playback")
```

Comment:
Will this warning identify which subclass instantiated the plugin? If not, could we add that?

Reply:
this is not for subclasses but rather for imports from old namespace, its done under new so we dont log on import of the module (which happens all the time and would be spammy) but rather on usage of the old import

nothing should be importing from here other than older ovos-audio versions, it's a "just in case" thing